cmd/d2/watch.go

```diff
@@ -345,19 +346,23 @@ func (w *watcher) compileLoop(ctx context.Context) error {
 			w.pw = newPW
 		}
 
-		b, err := compile(ctx, w.ms, true, w.layoutPlugin, w.themeID, w.inputPath, w.outputPath, w.pw.Page)
+		b, _, err := compile(ctx, w.ms, w.layoutPlugin, w.themeID, w.inputPath, w.outputPath, w.bundle, w.pw.Page)
+		errs := ""
 		if err != nil {
-			err = fmt.Errorf("failed to %scompile: %w", recompiledPrefix, err)
-			w.ms.Log.Error.Print(err)
-			w.broadcast(&compileResult{
-				Err: err.Error(),
-			})
+			if len(b) > 0 {
```

Comment:
can't you do the same in other place? (instead of written bool)

Reply:
what is `b` here? 😂

Reply:
bytes

Reply:
it was more in the sense, inside this function it is called `svg` and the return type is `bytes` so there is a lot context lost in just `b`.

Reply:
@alixander No you can get a partial svg even if the png/output failed to write.

@ejulio-ts yea fair enough i renamed thanks